geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/HAEventWrapper.java

```diff
@@ -87,16 +87,13 @@
       AtomicLongFieldUpdater.newUpdater(HAEventWrapper.class, "referenceCount");
 
   /**
-   * If true, the entry containing this HAEventWrapper instance will not be removed from the
-   * haContainer, even if the referenceCount value is zero.
+   * If greater than zero, the entry containing this HAEventWrapper instance will not be removed
+   * from the haContainer, even if the referenceCount value is zero.
    */
-  private transient boolean putInProgress = false;
-
-  /**
-   * A value true indicates that this instance is not used in the <code>haContainer</code>. So any
-   * changes in this instance will not be visible to the <code>haContainer</code>.
-   */
-  private transient boolean isRefFromHAContainer = false;
+  @SuppressWarnings("unused")
+  private transient volatile long putInProgressCount;
```

Comment:
Should this be volatile (for this atomic value as the reference count)?

Reply:
After reviewing the docs for this class, the answer is yes.  "A reflection-based utility that enables atomic updates to designated volatile long fields of designated classes".  
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicLongFieldUpdater.html

Also, note that the reason we use this over a regular AtomicLong is that it provides a slight performance benefit.